obvs/patchscope.py

```diff
@@ -181,14 +188,16 @@ def source_forward_pass(self) -> None:
 
         For each architecture, you need to know the name of the layers.
         """
-        with self.source_model.trace(self.source.text_prompt, remote=self.REMOTE) as _:
+        with self.source_model.trace(self.source.text_prompt, remote=self.REMOTE):
             if self.source.soft_prompt is not None:
                 # TODO: validate this with non GPT2 & GPTJ models
                 self.source_model.transformer.wte.output = self.source.soft_prompt
 
             self._source_hidden_state = self.manipulate_source().save()
             self.source_output = self.source_model.lm_head.output[0].save()
 
+        self._source_hidden_state = self._source_hidden_state.value
```

Comment:
Out of curiosity; why do we need this line?

Reply:
I need it when running nnsight with remote=True.

Without extracting the value tensor here, the target run will send the proxy object to the remote server, and nnsight doesn’t send the correct underlying value.

Maybe we should create an issue in nnsight 🤔

Reply:
Here is the nnsight issue: https://github.com/ndif-team/nnsight/issues/104

Reply:
I see. No further questions, let's merge.